client/src/models/Person.js

```diff
@@ -119,15 +120,24 @@ export default class Person extends Model {
         .label(Settings.fields.person.phoneNumber),
       endOfTourDate: yupDate
         .nullable()
-        .when("role", (role, schema) =>
+        .when(["role", "status", "uuid"], (role, status, uuid, schema) => {
```

Comment:
You're not returning anything (should return the schema). Conditions can be simplified, I'll have a go at it.